src/MapViewController.swift

```diff
@@ -846,10 +867,13 @@ open class MapViewController: UIViewController, LocationManagerDelegate {
     let oldAnnotations = self.currentAnnotations
     self.currentAnnotations = Dictionary()
     for (annotation, marker) in oldAnnotations {
-      let newMarker = TGMarker.init(mapView: self.tgViewController)
+      let newMarker = Marker.init()
+      addMarker(newMarker)
+      //TODO: also set polyline, polygon etc
       newMarker.point = marker.point
-      newMarker.stylingString = marker.stylingString
-      self.currentAnnotations[annotation] = newMarker
+      newMarker.tgMarker.stylingString = marker.stylingString
+      newMarker.tgMarker.stylingPath = marker.stylingPath
```

Comment:
`stylingString` and `stylingPath` are mutually exclusive. If you give a marker a styling string this one would be used, if you give it a path to a draw rule, the style of this targeted draw rule will be used. 

Refer implementation details in https://github.com/tangrams/tangram-es/pull/1242 and the notes in https://github.com/tangrams/tangram-es/blob/master/platforms/ios/src/TangramMap/TGMarker.h#L67.